src/sentry/utils/services.py

```diff
@@ -84,6 +82,44 @@ def expose(self, context):
             else:
                 context[key] = getattr(base, key)
 
+    def replace(self, instance=None, backend_path=None, options=None):
+        """
+        Replace this service with a new instance.
+
+        This is primarily useful for testing when you want to inject a different
+        backend.
+
+        Return a tuple of (``instance``, ``options``) for the replaced service.
+
+        >>> from sentry import servicename
+        >>> mockedservice = service.MockedService()
+        >>> original = servicename.replace(mockedservice)
+        >>> try:
+        >>>     # ...
+        >>> finally:
+        >>>     servicename.restore(*original)
+        """
+        rv = (self._wrapped, self._backend, self._options)
+        if backend_path and not instance:
```

Comment:
I feel like one (and only one) of these should be explicitly required.

Right now if you call the method with no explicit arguments this will succeed (but bind the instance as `None`, which makes no sense.)